sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosSchedulers.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation;
+
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+public class CosmosSchedulers {
+    private final static String COSMOS_PARALLEL_THREAD_NAME =  "cosmos-parallel";
+
+    // Using a custom parallel scheduler to be able to schedule retries etc.
+    // without being vulnerable to scenarios where applications abuse the
+    // Parallel scheduler and cause thread starvation on Reactor Core parallel scheduler
+    public final static Scheduler Parallel = Schedulers.newParallel(
+        COSMOS_PARALLEL_THREAD_NAME,
+        Schedulers.DEFAULT_POOL_SIZE,
```

Comment:
How it is different that default parallels scheduler , isn't it the exact same, are we doing it just for "cosmos-parallel" name ?

Reply:
No - we are doing it to avoid that customer's app can "steal" threads from the parallel thread-pool we are using b abusing it (for example doing a Thread.sleep on a parallel thread. When using the Reacor.Core built-in parallel thread the tasts we execute on it compete with other tasks the customer is scheduling on it. These changes avoid that.